src/main/kotlin/app/batch/DecryptionProcessor.kt

```diff
@@ -14,27 +14,27 @@ import org.springframework.stereotype.Component
 
 @Component
 class DecryptionProcessor(private val cipherService: CipherService,
-                          private val keyService: KeyService):
+                          private val keyService: KeyService) :
         ItemProcessor<SourceRecord, JsonObject> {
 
     @Throws(DataKeyServiceUnavailableException::class)
     override fun process(item: SourceRecord): JsonObject? {
         try {
             logger.info("Processing '$item'.")
-            val decryptedKey = keyService.decryptKey(item.encryption.encryptionKeyId,
-                                                           item.encryption.encryptedEncryptionKey)
+            val decryptedKey = keyService.decryptKey(
+                    item.encryption.encryptionKeyId,
+                    item.encryption.encryptedEncryptionKey)
             val decrypted =
-                    cipherService.decrypt(decryptedKey,
-                                                item.encryption.initializationVector,
-                                                item.dbObject)
+                    cipherService.decrypt(
+                            decryptedKey,
+                            item.encryption.initializationVector,
+                            item.dbObject)
             val jsonObject = Gson().fromJson(decrypted, JsonObject::class.java)
             jsonObject.addProperty("timestamp", item.hbaseTimestamp)
             return jsonObject
-        }
-        catch (e: DataKeyServiceUnavailableException) {
+        } catch (e: DataKeyServiceUnavailableException) {
             throw e
```

Comment:
Why catch this if we are just throwing it? I don't know enough about Kotlin to know how it handles errors but will this cut the stack trace off? Or is that the point...to hide it?

Reply:
Maybe at least do a logger.error() here?